independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java

```diff
@@ -80,6 +80,16 @@ static DecoratorInfo createDecorator(ClassInfo decoratorClass, BeanDeployment be
                 throw new DefinitionException("The delegate type " + delegateInjectionPoint.getRequiredType()
                         + " does not implement the decorated type: " + decoratedType);
             }
+            for (BuiltinBean bean : BuiltinBean.values()) {
+                if (bean.equals(BuiltinBean.RESOURCE)) {
+                    // do not take Resource into consideration
+                    continue;
+                }
+                if (bean.hasRawTypeDotName(decoratedType.name())) {
+                    throw new UnsupportedOperationException("Decorating built-in bean types is not supported! " +
```

Comment:
Just out of curiosity why `UnsupportedOperationException`? I would expect ISE or something like that... Although any RuntimeException is OK.

Reply:
My line of thought was that this is technically possible but we just don't support it. But I am fine with any exception really.